primitives/parachain-inherent/src/lib.rs

```diff
@@ -68,3 +69,40 @@ pub struct ParachainInherentData {
 	/// this means `sent_at` is **strictly** greater than the previous one (if any).
 	pub horizontal_messages: BTreeMap<ParaId, Vec<InboundHrmpMessage>>,
 }
+
+/// This struct provides ability to extend a message queue chain (MQC) and compute a new head.
+///
+/// MQC is an instance of a [hash chain] applied to a message queue. Using a hash chain it's
+/// possible to represent a sequence of messages using only a single hash.
+///
+/// A head for an empty chain is agreed to be a zero hash.
+///
+/// [hash chain]: https://en.wikipedia.org/wiki/Hash_chain
+#[derive(Default, Clone, codec::Encode, codec::Decode, scale_info::TypeInfo)]
+pub struct MessageQueueChain(RelayHash);
```

Comment:
Imo the docs should mention that `extend_hrmp` and `extend_dmp` should be used together.

Reply:
You mean to say that they should _not_ be used together right? My understanding is that an instance of `MessageQueueChain` is used to track either DMP from the relay chain or HRMP across a channel. But a given instance is never used to track both. Have I understood it correctly?

Reply:
Your understanding is right, I somehow forgot the word "not" :grimacing:

Reply:
addressed in 8933a4f9